### PR TITLE
Updated the file as per Guidelines. 

### PR DESCRIPTION
**Pull request by mishra-tanay**
Added Problem Statement .
Removed usage of class.
Mention expected time and space complexity

### CODE REVIEW

File: C++14/dp/longestValidParentheses.cpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+int longestValidParentheses(string s) {
+    int n=s.length();
+    if(n<2)
+    {
+        return 0;
+    }
+    int dp[n];
+    int max1=0;
+    for(int i=0;i<n;i++){
+    dp[i]=0;
```
Comment (likecs) [inline C++14/dp/longestValidParentheses.cpp]: fix indentation.

File: C++14/dp/longestValidParentheses.cpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            {
+                if(i-2>=0)
+                dp[i]=dp[i-2]+2;
+                else
+                dp[i]=2;
+            }
+            else
+            {
+                int ind=i-dp[i-1]-1;
+                cout<<i<<' '<<ind<<'\n';
```
Comment (likecs) [inline C++14/dp/longestValidParentheses.cpp]: remove this debug line.

File: C++14/dp/longestValidParentheses.cpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                if(ind>=0&& s[ind]=='(')
+                {
+                     if(ind-1>=0)
+                     {
+                         dp[i]=dp[i-1]+dp[ind-1]+2;
+                     }
+                      else
+                        {
+                        dp[i]=dp[i-1]+2;
+                        }
```
Comment (likecs) [inline C++14/dp/longestValidParentheses.cpp]: fix indentation for {}

**Comment by likecs** [comment]
Thanks for the changes. Minor comments before getting this merged.

Please rename the file using snake case than camel case so that it is consistent everywhere.

File: C++14/dp/longestValidParentheses.cpp (hunk 1 above)
Comment (mishra-tanay) [reply C++14/dp/longestValidParentheses.cpp]: Done

File: C++14/dp/longestValidParentheses.cpp (hunk 2 above)
Comment (mishra-tanay) [reply C++14/dp/longestValidParentheses.cpp]: Done

File: C++14/dp/longestValidParentheses.cpp (hunk 3 above)
Comment (mishra-tanay) [reply C++14/dp/longestValidParentheses.cpp]: Done

**Pull request merged**